SOS/SOS/APP/APP.c

```
@@ -1,6 +1,108 @@
-﻿/*
- * APP.c
- *
- * Created: 31/05/2023 03:08:53 م
- *  Author: click
- */ 
+﻿/**************************************************/
+/* Author  : Mustafa Mohammed Abdou               */
+/* Date    : 31 May , 2023                        */
+/* Version : V01                                  */
+/* Hint    : implementation of APP APIs           */
+/**************************************************/
+
+/** INCLUDE LIBRARIES **/
+#include "../SERVICE/common_macros.h"
+#include "../SERVICE/standard_types.h"
+
+/** INCLUDE LOWER LAYER FILES **/
+#include "../HAL/led/LED_interface.h"
+#include "../HAL/led/LED_config.h"
+#include "../HAL/button/button_interface.h"
+#include "../HAL/button/button_config.h"
+#include "../MCAL/EXTINT/INT_interface.h"
+#include "../OS/SOS_interface.h"
+
+/** INCLUDE LAYER FILES **/
+#include "APP.h"
+
+uint8_t button1_status   ; /** GLOBAL VARIABLE FOR BUTTON 1 **/
+
+uint8_t button2_status  ; /** GLOBAL VARIABLE FOR BUTTON 2 **/
+
+/** LED1 TOGGLE TASK **/
+void LED1_task()
```

All APIs should return an error state